File: dotnet/Lexy/Lexy.Compiler/Language/Expressions/Functions/MinutesFunction.cs
using Lexy.Compiler.Parser;

namespace Lexy.Compiler.Language.Expressions.Functions
{
    public class MinutesFunction : EndStartDateFunction
    {
        public const string Name = "MINUTES";

        protected override string FunctionName => Name;

        private MinutesFunction(Expression endDateExpression, Expression startDateExpression, SourceReference reference)
            : base(endDateExpression, startDateExpression, reference)
        {
        }

        public static BuiltInFunction Create(SourceReference reference, Expression endDateExpression,
            Expression startDateExpression) =>
            new MinutesFunction(endDateExpression, startDateExpression, reference);
    }
}

File: typescipt/src/language/expressions/functions/minutesFunction.ts


export class MinutesFunction extends EndStartDateFunction {
   public const string Name = `MINUTES`;

   protected override string FunctionName => Name;

   constructor(endDateExpression: Expression, startDateExpression: Expression, reference: SourceReference)
     : base(endDateExpression, startDateExpression, reference) {
   }

   public static ExpressionFunction Create(SourceReference reference, Expression endDateExpression,
     Expression startDateExpression) {
     return new MinutesFunction(endDateExpression, startDateExpression, reference);
   }
}
